app/javascript/components/helpers/helpers.js

```diff
@@ -18,10 +18,17 @@ export const backHandling = () => {
 export const rangeFormat = (tp) => {
   if (isBlank(tp)) return null
 
-  const start_date = new Date(tp.start_date)
-  const end_date = new Date(tp.end_date)
+  const baseDate = new Date(tp.start_date)
+
+  const start_date = new Date(baseDate)
+  start_date.setDate(baseDate.getDate() - (baseDate.getDay() - 1)) // move back to Monday
+
+  const end_date = new Date(start_date)
+  end_date.setDate(start_date.getDate() + 4) // move forward to Friday
```

Comment:
Please refactor so that we're getting `start_date` and `end_date` from the backend, and have `time_period#work_week_range` also get it from the same method. Having to change this functionality in 2 different places is a fundamental problem that we should fix.